### PR TITLE
feat: request/response updates for create invitation controller api 

### LINKED ISSUE #552: RestAPI - Get label value in CreateInvitation
Based on [TODO](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/restapi/operation/didexchange/didexchange.go#L94).

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- added request/response params to create invitation controller rest endpoint
- Part of #552 

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: test/bdd/agent_controller_steps.go [hunk 1]
```diff
@@ -113,6 +114,11 @@ func (a *AgentWithControllerSteps) createInvitation(inviterAgentID string) error
 		return fmt.Errorf("failed to get valid payload from create invitation for agent [%s]", inviterAgentID)
 	}
 
+	// verify result
+	if result.Payload.Label != label {
+		return fmt.Errorf("invitation label missmatch, expected[%s] but got [%s]", label, result.Payload.Label)
```
Comment (DRK3) [inline test/bdd/agent_controller_steps.go]: Typo: Should be "mismatch".

Comment (sudeshrshetty) [reply test/bdd/agent_controller_steps.go]: done

File: pkg/restapi/operation/didexchange/didexchange.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ const (
 	acceptExchangeRequest   = operationID + "/{id}/accept-request"
 	removeConnection        = operationID + "/{id}/remove"
 	connectionsWebhookTopic = "connections"
+	defLabel                = "agent"
```
Comment (troyronda) [inline pkg/restapi/operation/didexchange/didexchange.go]: is this the same default as aca py?

File: pkg/restapi/operation/didexchange/models/invitation.go [hunk 3]
```diff
@@ -24,6 +24,31 @@ type GenericError struct {
...
+//
+// This is used for creating invitation
+//
+type CreateInvitationParams struct {
+
+	// The Alias to be used in invitation to be created
+	Alias string `json:"alias"`
+
+	// The Public flag to create an invitation from the public DID
+	Public bool `json:"public,string,omitempty"`
```
Comment (troyronda) [inline pkg/restapi/operation/didexchange/models/invitation.go]: are these parameters in aca py?

Comment (sudeshrshetty) [reply pkg/restapi/operation/didexchange/models/invitation.go]: Yes, they have one more field called “auto” to auto accept requests for this connection. We don’t have that feature yet so I didn’t add this

File: pkg/restapi/operation/didexchange/models/invitation.go [hunk 4]
```diff
@@ -24,6 +24,31 @@ type GenericError struct {
...
+}
+
+// CreateInvitationParams model
+//
+// This is used for creating invitation
+//
+type CreateInvitationParams struct {
+
+	// The Alias to be used in invitation to be created
+	Alias string `json:"alias"`
```
Comment (troyronda) [inline pkg/restapi/operation/didexchange/models/invitation.go]: Is Alias the same as Label?

Comment (troyronda) [reply pkg/restapi/operation/didexchange/models/invitation.go]: I'm confused by this code: https://github.com/hyperledger/aries-framework-go/pull/614/files#diff-5213b774cf71e1cef26c7bb7c48ee94eR110

File: pkg/restapi/operation/didexchange/didexchange.go [hunk 5]
```diff
@@ -95,9 +96,22 @@ type Operation struct {
...
+	err := getQueryParams(&request, req.URL.Query())
+	if err != nil {
+		c.writeGenericError(rw, err)
+		return
+	}
+
+	// use label from request
+	label := defLabel
+	if request.CreateInvitationParams != nil && request.CreateInvitationParams.Alias != "" {
+		label = request.Alias
```
Comment (troyronda) [inline pkg/restapi/operation/didexchange/didexchange.go]: Why isn't this request.Label? I think I'm missing something.

File: pkg/restapi/operation/didexchange/didexchange.go (hunk 2 above)
Comment (sudeshrshetty) [reply pkg/restapi/operation/didexchange/didexchange.go]: removed

File: pkg/restapi/operation/didexchange/models/invitation.go (hunk 4 above)
Comment (sudeshrshetty) [reply pkg/restapi/operation/didexchange/models/invitation.go]: Removed that code

**Pull request merged**